doc/man3/EVP_PKEY_asn1_get_count.pod

```diff
@@ -0,0 +1,72 @@
+=pod
+
+=head1 NAME
+
+EVP_PKEY_asn1_find,
+EVP_PKEY_asn1_find_str,
+EVP_PKEY_asn1_get_count,
+EVP_PKEY_asn1_get0,
+EVP_PKEY_asn1_get0_info
+- enumerate public key ASN.1 methods
+
+=head1 SYNOPSIS
+
+ #include <openssl/evp.h>
+
+ int EVP_PKEY_asn1_get_count(void);
+ const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_get0(int idx);
+ const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find(ENGINE **pe, int type);
+ const EVP_PKEY_ASN1_METHOD *EVP_PKEY_asn1_find_str(ENGINE **pe,
+                                                    const char *str, int len);
+ int EVP_PKEY_asn1_get0_info(int *ppkey_id, int *pkey_base_id,
+                             int *ppkey_flags, const char **pinfo,
+                             const char **ppem_str,
+                             const EVP_PKEY_ASN1_METHOD *ameth);
+
+=head1 DESCRIPTION
+
+EVP_PKEY_asn1_count() returns a count of the number of public key
+ASN.1 methods available: it includes standard methods and any methods
+added by the application.
+
+EVP_PKEY_asn1_get0() returns the public key ASN.1 method B<idx>.
+The value of B<idx> must be between zero and EVP_PKEY_asn1_get_count()
+- 1.
+
+EVP_PKEY_asn1_find() looks up the B<EVP_PKEY_ASN1_METHOD> with NID
+B<type>.
+It searches through the methods added by the application first, and
+then through the standard methods.
```

Comment:
and what happens if `pe != NULL` ?

Reply:
that looks like the engine always wins ?

Reply:
in EVP_PKEY_asn1_find the first loop changes "type" if ASN1_PKEY_ALIAS
is in app_methods, and then if pe != NULL
ENGINE_get_pkey_asn1_meth_engine can look up something,
otherwise "t" from the frst loop is returned.
Weird...

Reply:
If I understand you correct, then it should not matter where we search first.
In that case this sentence can be removed.
However I wonder why the engine search uses the aliases at all, maybe the
whole "if (pe)" block should move up?

Reply:
previous comment refers to EVP_PKEY_asn1_find

Reply:
The loop at the start is there to follow aliases down to the root identity, which means that the methods provided by engines can't be aliases.  An engine can of course still provide an alias by adding them directly with `EVP_PKEY_asn1_add_alias`, but that will end up in the `app_methods` stack anyway...  So the loop to resolve aliases should come first, as it does.

Reply:
Maybe the documentation should reflect that ?

Reply:
Dunno how it would here.  `EVP_PKEY_asn1_add_alias` is the only way to add an alias, and it's not ENGINE specific, so...

The ENGINE API documentation needs to be refreshed for sure, though.  That's for a different PR...

Reply:
More or less make one sentence out of your previous two comments.
I believe the rationale is not obvious.